android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java

```diff
@@ -217,136 +236,111 @@ public void setConfiguration(BitmovinPlayerView view, ReadableMap config) throws
-
-        sourceConfiguration.addSourceItem(sourceItem);
-        // System.out.println("Created configuration with sourceItem: " + sourceConfiguration.getFirstSourceItem().getDashSource().getUrl());
-        _player.load(sourceConfiguration);
+        SourceConfiguration sourceConf = new SourceConfiguration();
+        sourceConf.addSourceItem(sourceItem);
+
+        // PT - Subtitle settings load correctly if this call is delayed, but the video does not
+        //      play properly - only audio plays and only going fullscreen fixes the video. This
+        //      behavior is different from the BasicPlayback example.
+        //
+        // Methods tested (all failed):
+        //
+        // 1.   Handler handler = new Handler(Looper.myLooper());
+        //      handler.post(() -> { _player.load(sourceConf); }); // also postDelayed
+        //
+        // 2.   final Timer timer = new Timer();
+        //      timer.schedule(new TimerTask() {
+        //          public void run() {
+        //              _reactContext.getCurrentActivity().runOnUiThread(() -> { _player.load(sourceConf); });
+        //          }
+        //      },100);
+        //
+        // 3.   _decorView.post(() -> { _player.load(sourceConf); });
+        _player.load(sourceConf);
     }
 
-    private void handleFullscreen(boolean fullscreen)
-    {
-        // System.out.println("XXX: handleFullscreen. " + fullscreen);
+    private void handleFullscreen(boolean fullscreen) {
         this._fullscreen = fullscreen;
+        // Fullscreen seems to work even without calling doSystemUiVisibility and doLayoutChanges
         this.doSystemUiVisibility(fullscreen);
         this.doLayoutChanges(fullscreen);
     }
 
-    private void doRotation(boolean fullScreen)
-    {
-        int rotation = this._activity.getWindowManager().getDefaultDisplay().getRotation();
-
-        if (fullScreen)
-        {
-            // System.out.println("XXX: doRotation fullscreen.");
-            switch (rotation)
-            {
-                case Surface.ROTATION_270:
-                    this._activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE);
-                    break;
-
-                default:
-                    this._activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
-            }
-        }
-        else
-        {
-            // System.out.println("XXX: doRotation no fullscreen.");
-            this._activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
-        }
-    }
-
-    private void doSystemUiVisibility(final boolean fullScreen)
-    {
-        this._decorView.post(new Runnable()
-        {
-            @Override
-            public void run()
-            {
-                int uiParams = FullscreenUtil.getSystemUiVisibilityFlags(fullScreen, true);
-                _decorView.setSystemUiVisibility(uiParams);
-            }
+    //private void doRotation(boolean fullScreen) {
```

Comment:
Was doRotation() not used?

Reply:
Correct

Reply:
I started doing native full screen based on the examples but found a way to rotate using react-native side.
System Visibility affects different android versions and tablets so I left it in there (takes away system nav bars etc).